### DISCUSSION TITLE
Joyride dependancy error. TypeError: Invalid parameter: element must be an HTMLElement

### DISCUSSION THREAD

**Discussion by Stivifortes** [gilbarbara/react-joyride#708] [Q&A]
When I first start the tour, I get these error : TypeError: Invalid parameter: element must be an HTMLElement

Scope:
  ```if (!(_element instanceof HTMLElement)) {
    throw new TypeError('Invalid parameter: element must be an HTMLElement');
  }```

``` if (changed('lifecycle', LIFECYCLE.TOOLTIP)) {
        callback(_objectSpread2(_objectSpread2({}, state), {}, {
          step: step,
          type: EVENTS.TOOLTIP
        }));
        this.scope = new Scope(this.tooltip, {
          selector: '[data-action=primary]'
        });
        this.scope.setFocus();
      }
```

````  instance.componentDidUpdate(prevProps, prevState, instance.__reactInternalSnapshotBeforeUpdate);
            stopPhaseTimer();
          }
````

````
 if (effectTag & (Update | Callback)) {
      recordEffect();
      var current = nextEffect.alternate;
      commitLifeCycles(root, current, nextEffect);
    }
````

````
if (typeof window.event !== 'undefined' && window.hasOwnProperty('event')) {
          window.event = windowEvent;
        }

        func.apply(context, funcArgs);
        didError = false;
      } 
````

**Comment by jiyang-sas** [comment]
Sometimes I do get this error if I set the auto start flag run to be true.  I get this error sometimes, not all the times. I debugged into the code and it looks like the tooltip component is null,  in this case it is the intro step where no target element is specfied.  The following code is inside the Index.js and this.tooltip is null. 

      if (changedTo('lifecycle', LIFECYCLE.TOOLTIP)) {
        callback(_objectSpread2({}, state, {
          step: step,
          type: EVENTS.TOOLTIP
        }));
        this.scope = new Scope(this.tooltip, {
          selector: '[data-action=primary]'
        });
        this.scope.setFocus();
      }

eventually the error is throw at :

  if (!(_element instanceof HTMLElement)) {
    throw new TypeError('Invalid parameter: element must be an HTMLElement');
  }

I don't why the intro step tooltip is sometime null.
